README.md

```diff
@@ -80,24 +79,181 @@ After you input your application credentials, you can click the setup button, wh
 
-Once that setup has been completed, you can very easily call all the necessary API requests to connect to a cloudlet with your application deployed. Here is some example code using the MobiledgeXIntegration class that comes with the package : 
+Once that setup has been completed, you can very easily call all the necessary API requests to connect to a cloudlet with your application deployed. Here is some example code using the MobiledgeXIntegration class that comes with the package 
+
+
+**Getting Edge Connection Url**
+
+
 
 ```csharp
 using MobiledgeX;
+using DistributedMatchEngine;
+using UnityEngine;
+using System.Collections;
+
+[RequireComponent(typeof(MobiledgeX.LocationService))]
+public class NetworkManager : MonoBehaviour
+{ 
+    IEnumerator Start()
+    {
+        yield return StartCoroutine(MobiledgeX.LocationService.EnsureLocation()); // Location is needed to connect you to the closet edge
+        GetEdgeConnection();
+    }
+     async void GetEdgeConnection()
+    {
+        MobiledgeXIntegration mxi = new MobiledgeXIntegration();
+        try
+        {
+            await mxi.RegisterAndFindCloudlet();
+        }
+        catch(DmeDnsException)
+        {
+            mxi.UseWifiOnly(true); // if you carrier is not supported yet, WifiOnly will connect you to the closet MobiledgeX Public Cloud
+            await mxi.RegisterAndFindCloudlet();
+        }
+
+        mxi.GetAppPort(LProto.L_PROTO_HTTP); // Get the port of the desired protocol
+        string url = mxi.GetUrl("http"); // Get the url of the desired protocol
+    }
+    
+}
+```
+
+
+MobiledgeX requires Location and Mobile phone carrier information (specifically MccMnc code)
+
+
+If you carrier is not supported yet by MobiledgeX it will throw a DmeDnsException and it will connect you to the closet MobiledgeX Public Cloud 
+
+
+
+**In UnityEditor** 
+
 
-MobiledgeXIntegration integration = new MobiledgeXIntegration();
-bool registered = await integration.Register(); //calls Register Client
-DistributedMatchEngine.FindCloudletReply findCloudletReply = await integration.FindCloudlet(); //calls Find Cloudlet
 
-MatchingEngine dme = integration.me;
+(Location is not used) there is fallback Location in [RunTime/Scripts/MobiledgeXIntegration.cs](https://github.com/mobiledgex/edge-cloud-sdk-unity/blob/master/Runtime/Scripts/MobiledgeXIntegration.cs) , You change it to your desired test location.
 
-Dictionary<int, AppPort> appPortsDict = dme.GetHTTPAppPorts(findCloudletReply);
-int public_port = findCloudletReply.ports[0].public_port; // if you only have one port
-AppPort appPort = appPortsDict[public_port];
-HttpClient http = await dme.GetHTTPClient(findCloudletReply, appPort, public_port, 5000);
-HttpResponseMessage message = await http.GetAsync("/"); //makes a get request
+Since there is no phone carrier information in UnityEditor, by default you will be using WifiOnly mode which will connect you to the closet MobiledgeX Public Cloud to the fallback location.
```

Comment:
more accurately you will connect to the wifi DME

Reply:
Do we have any explanation about wifi DME , I can link it to

by default you will be using WifiOnly mode which will connect you [wifi DME](link explaining wifi dme)

Reply:
we have this : https://developers.mobiledgex.com/getting-started/connecting-client-app#distributed-matching-engine